extensions/interactions/FractionInput/FractionInput.py

```diff
@@ -78,3 +80,374 @@ class FractionInput(base.BaseInteraction):
+            proto object.
+        """
+        is_exactly_equal_to_proto = (
+            state_pb2.FractionInputInstance.RuleSpec.IsExactlyEqualToSpec(
+                input=cls._to_fraction_proto(fraction)))
+
+        return is_exactly_equal_to_proto
+
+    @classmethod
+    def _to_is_equivalent_to_proto(cls, fraction):
+        """Creates a proto object for IsEquivalentToSpec.
+
+        Args:
+            fraction: Fraction. The fraction domain object.
+
+        Returns:
+            IsEquivalentToSpec. The IsEquivalentToSpec
+            proto object.
+        """
+        equivalent_to_proto = (
+            state_pb2.FractionInputInstance.RuleSpec.IsEquivalentToSpec(
+                input=cls._to_fraction_proto(fraction)))
+
+        return equivalent_to_proto
+
+    @classmethod
+    def _to_is_equivalent_to_and_in_simplest_form_proto(cls, fraction):
+        """Creates a proto object for IsEquivalentToAndInSimplestFormSpec.
+
+        Args:
+            fraction: Fraction. The fraction domain object.
+
+        Returns:
+            IsEquivalentToAndInSimplestFormSpec. The
+            IsEquivalentToAndInSimplestFormSpec proto object.
+        """
+        is_equivalent_to_and_in_simplest_form_proto = (
+            state_pb2.FractionInputInstance
+            .RuleSpec.IsEquivalentToAndInSimplestFormSpec(
+                input=cls._to_fraction_proto(fraction)))
+
+        return is_equivalent_to_and_in_simplest_form_proto
+
+    @classmethod
+    def _to_is_less_than_proto(cls, fraction):
+        """Creates a proto object for IsLessThanSpec.
+
+        Args:
+            fraction: Fraction. The fraction domain object.
+
+        Returns:
+            IsLessThanSpec. The IsLessThanSpec
+            proto object.
+        """
+        is_less_than_proto = (
+            state_pb2.FractionInputInstance.RuleSpec.IsLessThanSpec(
+                input=cls._to_fraction_proto(fraction)))
+
+        return is_less_than_proto
+
+    @classmethod
+    def _to_is_greater_than_proto(cls, fraction):
+        """Creates a proto object for IsGreaterThanSpec.
+
+        Args:
+            fraction: Fraction. The fraction domain object.
+
+        Returns:
+            IsGreaterThanSpec. The IsGreaterThanSpec
+            proto object.
+        """
+        is_greater_than_proto = (
+            state_pb2.FractionInputInstance.RuleSpec.IsGreaterThanSpec(
+                input=cls._to_fraction_proto(fraction)))
+
+        return is_greater_than_proto
+
+    @classmethod
+    def _to_has_numerator_equal_to_proto(cls, f):
+        """Creates a proto object for HasNumeratorEqualToSpec.
+
+        Args:
+            f: Fraction. The fraction domain object.
+
+        Returns:
+            HasNumeratorEqualToSpec. The HasNumeratorEqualToSpec
+            proto object.
+        """
+        has_numerator_equal_to_proto = (
+            state_pb2.FractionInputInstance
+            .RuleSpec.HasNumeratorEqualToSpec(
+                input=f))
```

Comment:
Why do we do `cls._to_fraction_proto(fraction)` above and not here?

Reply:
Also, all these methods are pretty similar, can they also be generalized using some kind of mapping?

Reply:
```
    message HasNumeratorEqualToSpec {
      option (structure_proto_version_type) = STATE_PROTO_VERSION;

      // Corresponds to the creator-specified numerator to match against the user's answer.
      int32 input = 1;
    }
```

These rules takes input as int
- `HasNumeratorEqualToSpec`
- `HasDenominatorEqualToSpec`
- `HasIntegerPartEqualToSpec`

Othe takes fraction domain object.